ee/control/control.go

```diff
@@ -35,6 +35,7 @@ type ControlService struct {
 	lastFetched          map[string]string
 	consumers            map[string]consumer
 	subscribers          map[string][]subscriber
+	knownSubsystems      map[string]struct{}
```

Comment:
Would `len(subscribers[...]) > 0` have been the same?

Reply:
I don't think so, though I might be misinterpreting your suggestion -- there can be a valid subsystem that has a consumer but no subscribers.

Reply:
`len` of both?

Reply:
I'm saying that we probably don't need `knownSubsystems`, we should be able to derive it from `consumers` and `subscribers`. And that this is infrequent enough there's no performance win in caching it. 

It's also pretty harmless, but I started by wanting to quibble about names, and then I realized we didn't need it at all.

Reply:
I'll update in a new PR